examples/plot_wlasso.py

```diff
@@ -10,90 +10,93 @@
-#
+#          Mathurin Massias
+#          Kenan Sehic
 # License: BSD (3-clause)
 
-import time
-
 import numpy as np
 
-from sklearn.datasets import make_regression
-from sklearn.model_selection import train_test_split
+from celer import Lasso, LassoCV
 
-from celer import Lasso
+from sklearn.datasets import make_regression
 
 from sparse_ho.models import wLasso
 from sparse_ho.criterion import CV
 from sparse_ho.implicit_forward import ImplicitForward
 from sparse_ho.utils import Monitor
 from sparse_ho.ho import grad_search
-from sparse_ho.datasets import get_data
 
-print(__doc__)
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error
 
-dataset = 'rcv1_train'
-# dataset = 'simu'
+# from wlasso_hpo import sparse_ho_model, random_search
 
-if dataset != 'simu':
-    X_train, X_val, X_test, y_train, y_val, y_test = get_data(dataset)
-    X_train = X_train[:, :1000]
-    X_test = X_test[:, :1000]
-else:
-    X, y = make_regression(n_samples=100, n_features=100, noise=1)
-    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33)
-    X_train, X_val, y_train, y_val = train_test_split(
-        X_train, y_train, test_size=0.5)
 
-n_samples, n_features = X_train.shape
+X, y = make_regression(n_samples=500, n_features=600, noise=40, random_state=2)
+X_train_val, X_test, y_train_val, y_test = train_test_split(
+    X, y, test_size=0.15)
 
-print("Starting path computation...")
-n_samples = len(y_train)
-alpha_max = np.max(np.abs(X_train.T.dot(y_train))) / X_train.shape[0]
-log_alpha0 = np.log(alpha_max / 10)
+all_indices = list(range(X_train_val.shape[0]))
+train_indices, val_indices = train_test_split(
+    all_indices, test_size=0.5, random_state=42)
 
-n_alphas = 10
-p_alphas = np.geomspace(1, 0.0001, n_alphas)
-alphas = alpha_max * p_alphas
-log_alphas = np.log(alphas)
+X_train = X_train_val[train_indices, :]
+X_val = X_train_val[val_indices, :]
 
-tol = 1e-7
-max_iter = 1e5
+y_train = y_train_val[train_indices]
+y_val = y_train_val[val_indices]
 
-##############################################################################
-# Grid-search
-# -----------
 
-# The sklearn solver is indeed very long on the considered problems!
-# estimator = linear_model.Lasso(
-#     fit_intercept=False, max_iter=1000, warm_start=True)
+# max penalty value
+alpha_max = np.max(np.abs(X_train.T.dot(y_train))) / X_train.shape[0]
+
+n_alphas = 30
+alphas = alpha_max * np.geomspace(1, 0.001, n_alphas)
 
-# celer is way faster !
-# https://github.com/mathurinm/celer
+train_split_cv = [train_indices]
+val_split_cv = [val_indices]
 
-estimator = Lasso(
-    fit_intercept=False, max_iter=1000, warm_start=True)
 
+# vanilla LassoCV
+print("========== Celer's LassoCV started ===============")
+model_cv = LassoCV(verbose=False, fit_intercept=False,
+                   alphas=alphas, tol=1e-7, max_iter=100,
+                   cv=zip(train_split_cv, val_split_cv)).fit(X_train_val,
```

Comment:
here is what I am talking about earlier

Reply:
could you detail a little bit more? I am not sure I understand